### PR TITLE
new: start migration to support error wrapping

### PR DESCRIPTION
**Pull request by primalmotion**
This patch improve the old manipulate.Error to support error wrapping. Old methods have been deprecated

### CODE REVIEW

File: errors_test.go [hunk 1]
```diff
@@ -11,134 +11,160 @@
 
 package manipulate
```
Comment (t00f) [inline errors_test.go:13]: Should we uncomment these tests?

File: internal/push/utils.go [hunk 2]
```diff
@@ -30,11 +30,11 @@ func decodeErrors(r io.Reader, encoding elemental.EncodingType) error {
 
 	data, err := ioutil.ReadAll(r)
 	if err != nil {
-		return manipulate.NewErrCannotUnmarshal(fmt.Sprintf("%s: %s", err.Error(), string(data)))
+		return manipulate.ErrCannotUnmarshal{Err: fmt.Errorf("%s: %s", err.Error(), string(data))}
 	}
 
 	if err := elemental.Decode(encoding, data, &es); err != nil {
-		return manipulate.NewErrCannotUnmarshal(fmt.Sprintf("%s: %s", err.Error(), string(data)))
+		return manipulate.ErrCannotUnmarshal{Err: fmt.Errorf("%s: %s", err.Error(), string(data))}
```
Comment (t00f) [inline internal/push/utils.go:37]: Should we unwrap errors here too?

**Review by t00f — CHANGES_REQUESTED** [review]
LGTM except tests that are commented

File: errors_test.go (hunk 1 above)
Comment (primalmotion) [reply errors_test.go:13]: oopsy

File: internal/push/utils.go (hunk 2 above)
Comment (primalmotion) [reply internal/push/utils.go:37]: yep

**Review by t00f — DISMISSED** [review]

**Pull request merged**